yarn-project/sequencer-client/src/test/l2-block-publisher.test.ts

```diff
@@ -97,50 +101,76 @@ describe.skip('L1Publisher integration', () => {
     return makeEmptyProcessedTxFromHistoricTreeRoots(historicTreeRoots);
   };
 
-  it('Build 2 blocks of 4 txs building on each other', async () => {
-    const newL1ToL2Messages = new Array(NUMBER_OF_L1_L2_MESSAGES_PER_ROLLUP).fill(new Fr(0n));
+  const makeContractDeployProcessedTx = async (seed = 0x1) => {
+    const tx = await makeEmptyProcessedTx();
+    tx.data.end.newContracts = [makeNewContractData(seed + 0x1000)];
+    return tx;
+  };
+
+  const makePrivateProcessedTx = async (seed = 0x1) => {
+    const tx = await makeEmptyProcessedTx();
+    tx.data.end.newCommitments = range(KERNEL_NEW_COMMITMENTS_LENGTH, seed + 0x100).map(fr);
+    tx.data.end.newNullifiers = range(KERNEL_NEW_NULLIFIERS_LENGTH, seed + 0x200).map(fr);
+    return tx;
+  };
 
+  const makePublicCallProcessedTx = async (seed = 0x1) => {
+    const publicTx = makePublicTx(seed);
+    const kernelOutput = KernelCircuitPublicInputs.empty();
+    kernelOutput.end.stateReads[0] = new PublicDataRead(fr(1), fr(0));
+    kernelOutput.end.stateTransitions[0] = new PublicDataTransition(fr(2), fr(0), fr(12));
+    kernelOutput.constants.historicTreeRoots = await getCombinedHistoricTreeRoots(builderDb);
+    return await makeProcessedTx(publicTx, kernelOutput, makeProof());
+  };
+
+  it('Build 2 blocks of 4 txs building on each other', async () => {
```

Comment:
Why not use the actual block builder here, rather than manually assembling the block? Seems like it'd lead to less duplication.

Reply:
Just good ol' debt.